File: Placement Sheet/Stack & Queue/MinStack.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var MinStack = /** @class */ (function () {
    function MinStack() {
        this.stack = [];
        this.minStack = [];
    }
    //Push...
    MinStack.prototype.push = function (data) {
        this.stack.push(data);
        if (this.minStack.length == 0 || data <= this.getMin()) {
            this.minStack.push(data);
        }
    };
    //Pop...
    MinStack.prototype.pop = function () {
        var poppedElm = this.stack.pop();
        if (poppedElm === this.getMin()) {
            this.minStack.pop();
        }
        return poppedElm;
    };
    //Peek...
    MinStack.prototype.peek = function () {
        return this.stack[this.stack.length - 1];
    };
    MinStack.prototype.getMin = function () {
        return this.minStack[this.minStack.length - 1];
    };
    return MinStack;
}());
var minStack = new MinStack();
minStack.push(5);
minStack.push(3);
minStack.push(7);
minStack.push(2);
console.log("Min: ".concat(minStack.getMin()));
minStack.pop();
console.log("Min: ".concat(minStack.getMin()));
console.log("Top: ".concat(minStack.peek()));

File: Placement Sheet/Stack & Queue/Service Based/Min_Stack/MinStack.java
package Min_Stack;

public class MinStack {
    private Stack stack;
    private Stack minStack;

    public MinStack(int capacity) {
        this.stack = new Stack(capacity);
        this.minStack = new Stack(capacity);
    }

    public void push(int num) {
        stack.push(num);

        if (minStack.isEmpty() || num <= minStack.top()) {
            minStack.push(num);
        } else {
            minStack.push(minStack.top());
        }
    }

    public int pop() {
        if (stack.isEmpty()) {
            return -1;
        }
        minStack.pop();
        return stack.pop();
    }

    public int top() {
        if (stack.isEmpty()) {
            return -1;
        }
        return stack.top();
    }

    public int getMin() {
        if (minStack.isEmpty()) {
            return -1;
        }
        return minStack.top();
    }
}
